packages/llamaindex/src/indices/BaseIndex.ts

```diff
@@ -61,6 +67,37 @@ export abstract class BaseIndex<T> {
     options?: Omit<ContextChatEngineOptions, "retriever">,
   ): BaseChatEngine;
 
+  /**
+   * Returns a query tool by calling asQueryEngine.
+   * Either options or retriever can be passed, but not both.
+   * If options are provided, they are passed to generate a retriever.
+   */
+  asQueryTool(
+    params: (
+      | {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          options: any;
```

Comment:
could we fix this any type?

Reply:
yes, it's ugly, but let's do it in another PR